lint/source.go

```diff
@@ -63,34 +62,35 @@ func newLocPath(p ...int) locPath {
 
 // buildLocPathMap creates a map of locPath to *descriptorpb.SourceCodeInfo_Location
 // from *descriptorpb.SourceCodeInfo.
-func buildLocPathMap(sci *descriptorpb.SourceCodeInfo) map[locPath]*descriptorpb.SourceCodeInfo_Location {
-	m := make(map[locPath]*descriptorpb.SourceCodeInfo_Location)
+func buildLocPathMap(sci protoreflect.SourceLocations) map[locPath]*protoreflect.SourceLocation {
+	m := make(map[locPath]*protoreflect.SourceLocation)
 	if sci == nil {
 		return m
 	}
 
-	for _, loc := range sci.GetLocation() {
-		var path []int
-		for _, v := range loc.GetPath() {
-			path = append(path, int(v))
+	for i := 0; i < sci.Len(); i++ {
+		loc := sci.Get(i)
+		path := make([]int, len(loc.Path))
+		for k := range path {
+			path[k] = int(loc.Path[k])
```

Comment:
I'm not sure if this is a nitpick but I got yelled at for this in go readability review. The recommended way is using append, so you could do something like:

```
path := make([]int, 0, len(loc.Path))
for k := range path {
  path = append(path, int(loc.Path[k]))
}
```

Not blocking on this just wanted to let you know.

Reply:
Make sense. It saves future mistakes that might result index out of bound errors. I will change it accordingly. Thanks!